Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -354,13 +527,54 @@ bool isFunctionJson(string fullName)
             }
         }
 
-        private async Task<string> DeployZipLocally(ZipDeploymentInfo zipDeploymentInfo, ITracer tracer)
+        //Used by OneDeploy to fetch artifacts that are not ZIP files. OneDeploy ZIP files are handled with 
+        //LocalZipHandler()
+        private async Task OneDeployFetch(IRepository repository, DeploymentInfoBase deploymentInfo, string targetBranch, ILogger logger, ITracer tracer)
+        {
+            var artifactDeploymentInfo = (ArtifactDeploymentInfo)deploymentInfo;
+            //If ArtifactURL property is defined, the artifact is to be fetched from ArtifactURL. Otherwise, 
+            //the artifact is to be read from the request's body. 
+            HttpContent content = !string.IsNullOrEmpty(artifactDeploymentInfo.RemoteURL)
+                ? await DeploymentHelper.GetArtifactContentFromURL(artifactDeploymentInfo, tracer)
+                : Request.Content;
+
+            var tempDirPath = repository.RepositoryPath;
+
+            var targetInfo = FileSystemHelpers.DirectoryInfoFromDirectoryName(tempDirPath);
+            if (targetInfo.Exists)
+            {
+
+                // If tempDirPath already exists, rename it so we can delete it later 
+                var moveTarget = Path.Combine(targetInfo.Parent.FullName, Path.GetRandomFileName());
+                using (tracer.Step(string.Format("Renaming extractTargetDirectory({0}) to tempDirectory({1})", targetInfo.FullName, moveTarget)))
+                {
+                    targetInfo.MoveTo(moveTarget);
+                }
+            }
+
+
+            var tempFilePath = Path.Combine(tempDirPath, deploymentInfo.TargetFileName);
+            Directory.CreateDirectory(tempDirPath);
+            using (tracer.Step("Saving request content to {0}", tempFilePath))
+            {
+                var copyTask = content.CopyToAsync(tempFilePath, tracer);
+
+                //Deletes all files and directories except for tempFilePath, tempDirPath, and tracer
+                var cleanTask = Task.Run(() => DeleteFilesAndDirsExcept(tempFilePath, tempDirPath, tracer));
+                await Task.WhenAll(copyTask, cleanTask);
```

Comment:
// Let the copy and cleanup tasks to run in parallel and wait for them to complete.